opm/utility/ECLGraph.hpp

```diff
@@ -0,0 +1,154 @@
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#ifndef OPM_ECLGRAPH_HEADER_INCLUDED
+#define OPM_ECLGRAPH_HEADER_INCLUDED
+
+#include <opm/core/props/BlackoilPhases.hpp>
+
+#include <cstddef>
+#include <memory>
+#include <vector>
+
+#include <boost/filesystem.hpp>
+
+/// \file
+///
+/// Facility for extracting active cells and neighbourship relations from
+/// on-disk ECLIPSE output, featuring on-demand and cached property loading
+/// from backing object (e.g., restart vectors at various time points).
+
+namespace Opm {
+
+    /// Package an ECLIPSE result set (represented as GRID, INIT, and
+    /// restart files) in form usable with the computational engine
+    /// represented by class \code Opm::FlowDiagnostics::ToolBox \endcode.
+    class ECLGraph
+    {
+    public:
+        using Path = boost::filesystem::path;
+
+        /// Disabled default constructor.
+        ECLGraph() = delete;
+
+        /// Destructor.
+        ~ECLGraph();
+
+        /// Move constructor.
+        ///
+        /// \param[in] rhs Graph from which to appropriate resources.
+        /// Invalid upon return.
+        ECLGraph(ECLGraph&& rhs);
+
+        /// Disabled copy constructor
+        ECLGraph(const ECLGraph& rhs) = delete;
+
+        /// Move assignment operator.
+        ///
+        /// \param[in] rhs Graph from which to appropriate resources.
+        ///
+        /// \return Reference to \code *this \endcode.
+        ECLGraph& operator=(ECLGraph&& rhs);
+
+        /// Disabled assignment operator.
+        ECLGraph& operator=(const ECLGraph& rhs) = delete;
+
+        /// Named constructor.
+        ///
+        /// \param[in] grid Name or prefix of ECL grid (i.e., .GRID or
+        ///                 .EGRID) file.
+        ///
+        /// \param[in] init Name of ECL INIT file corresponding to \p grid
+        ///                 input.  Assumed to provide at least a complete
+        ///                 set of pore-volume values (i.e., for all global
+        ///                 cells defined in the \p grid).
+        ///
+        ///                 If available in the INIT file, the constructor
+        ///                 will also leverage the transmissibility data
+        ///                 when constructing the active cell neighbourship
+        ///                 table.
+        ///
+        /// \return Fully formed ECLIPSE connection graph with property
+        /// associations.
+        static ECLGraph
+        load(const Path& grid, const Path& init);
```

Comment:
> This data is handled

Sounds good. So the result (except for possibly some connections with zero fluxes) will be the same regardless of transmissibility data being present or not (i.e. it's just used to improve performance and prune the graph)?